main.py

```diff
@@ -177,19 +188,34 @@ def main():
     states['CREATE_SAVE_CONTACT']['TRANSITIONS'] = []
     states['CREATE_SAVE_CONTACT']['TRANSITIONS'].append({'REGEXEN':['.*'],'DESTINATION':'ROOT'})
 
+    # CANCEL
     states['CANCEL_AND_ROOT'] = {'GREET':cancel_and_root,'STORE':'CONTACT_PHONE'}
     states['CANCEL_AND_ROOT']['TRANSITIONS'] = []
     states['CANCEL_AND_ROOT']['TRANSITIONS'].append({'REGEXEN':['.*'],'DESTINATION':'ROOT'})
 
-
+    # READ
     states['READ'] = {'GREET':show_list_of_contacts}
     states['READ']['TRANSITIONS'] = []
     states['READ']['TRANSITIONS'].append({'REGEXEN':['.*'],'DESTINATION':'ROOT'})
     
-
-    states['UPDATE'] = {}
+    # UPDATE
+    states['UPDATE'] = {'GREET':prompt_for_name,'STORE':'CONTACT_NAME'}
```

Comment:
Where is the function prompt_for_name defined?
Is it defined in another file?

Reply:
line 106. Reusing the same function for CREATE. keeping it DRY. 

```
def prompt_for_name():
    say("Please provide a name for the new contact:")
    # "Please provide a name:"
    # "Please provide a name for the contact:"
    # "Provide a name:"
```
The wording could use some work since this isn't a `new` contanct.

Reply:
Write another function

Reply:
I'd need to wrote one for the Update and Delete function as well. Is this still advisable?

Reply:
yes